src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Processors/Settings/GeneralizeSetting.cs

```diff
@@ -23,6 +23,15 @@ public static GeneralizeSetting CreateFromRuleSettings(Dictionary<string, object
             try
             {
                 cases = JObject.Parse(ruleSettings.GetValueOrDefault(RuleKeys.Cases)?.ToString());
+                
+                FhirPathCompiler compiler = new FhirPathCompiler();
+
+                // Validate cases for compiling.
+                foreach (var eachCase in cases)
+                {
+                    compiler.Compile(eachCase.Key.ToString());
```

Comment:
shall we save/cache the compile result?

Reply:
compiler already has cache inside.

Reply:
provide a doc ?

Reply:
After double check , compiler doesn't have cache. But we only run compiler once when validating config.

Reply:
do you think the compiling result can be used later?

Reply:
The expression will be cached when using Predicate function. After offline discussion,  we don't need to cache it here.